yo1k/tic_tac_toe/core/game.py

```diff
@@ -183,15 +186,26 @@ def __init__(self, action_queues: Sequence[ActionQueue]):
         self.__action_queues = action_queues
 
     def advance(self, state: State):
```

Comment:
Let's extract two methods, e.g., `advance_beginning_outround` and `advance_inround` from `advance` to simplify it. The idea is similar to that currently used in each top-level branch in the body of the `advance` method.

Reply:
Done.